processor/resourcedetectionprocessor/internal/kubeadm/config.go

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kubeadm // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm"
+
+import (
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/k8sconfig"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/kubeadm/internal/metadata"
+)
+
+type Config struct {
+	k8sconfig.APIConfig `mapstructure:",squash"`
+	ResourceAttributes  metadata.ResourceAttributesConfig `mapstructure:"resource_attributes"`
+	configMapName       string
+	configMapNamespace  string
```

Comment:
Is the user meant to be able to configure there values via `config.yaml`? If yes, we need to add a `mapstructure` value here as well as add to the README how to configure. If not, we need to fix error messages about these values not being set (if the user can't set them, the error messages should be more clear).

Reply:
currently the values are not configurable at all, it up to us if we want to have them configurable, I do not have a strong opinion about this

Reply:
If they're not configurable, I think we should remove them from `Config`, and remove the `config` prefix, that way we can have configurable options together.

Can we just set these in `NewProvider` where they're going to be used and referenced?

Reply:
the `config` prefix is due to that the information about the cluster name is available in a [configMap](https://kubernetes.io/docs/concepts/configuration/configmap/) on the local cluster. Therefore the prefix is part of the name of the object.

Adapted the code, so the default values (`configMap` name and namespace) are defined in the `kubeadm.go` and passed from this place further

Reply:
Good point on the prefix, my apologies. I was only thinking of the collector's use of `config` and didn't think of it in the context of k8s as we're using it here 👍